alternator/expressions.g

```diff
@@ -231,19 +250,20 @@ primitive_condition returns [parsed::primitive_condition c]:
 // common rule prefixes, and (lack of) support for operator precedence.
 // These rules could have been written more clearly using a more powerful
 // parser generator - such as Yacc.
-boolean_expression returns [parsed::condition_expression e]:
-	  b=boolean_expression_1       { $e.append(std::move($b.e), '|'); }
-	  (OR b=boolean_expression_1   { $e.append(std::move($b.e), '|'); } )*
+// See comment above why the "depth" counter was needed here.
+boolean_expression[int depth] returns [parsed::condition_expression e]:
+	  b=boolean_expression_1[depth]       { $e.append(std::move($b.e), '|'); }
+	  (OR b=boolean_expression_1[depth]   { $e.append(std::move($b.e), '|'); } )*
 	;
-boolean_expression_1 returns [parsed::condition_expression e]:
-	  b=boolean_expression_2       { $e.append(std::move($b.e), '&'); }
-	  (AND b=boolean_expression_2  { $e.append(std::move($b.e), '&'); } )*
+boolean_expression_1[int depth] returns [parsed::condition_expression e]:
+	  b=boolean_expression_2[depth]       { $e.append(std::move($b.e), '&'); }
```

Comment:
Yes, I sort of did this deliberately. The different levels boolean_expression, boolean_expression_1, boolean_expression_2, only needs to exist because ANTLR can't understand recursive expressions or operator precedence (I  tried to explain this unfortunate mess in a comment in expressions.g). So in the call trace you always see these three calling each other, e.g., "((" gets translated to calls boolean_expression(boolean_expression_1(boolean_expression_2(boolean_expression(boolean_expression_1(boolean_expression_2).
I just wanted to count this as "2" instead of "6", I thought it made it easier to understand what is really being limited for the user (the user thinks there is a depth of 2, not 6). So this is why I only incremented in the transition back to boolean_expression. 
Since I set the limit low (only 400), I don't think it matters if this really means depth 400 or 400*3 in the stack, both seem to be fine on our stack.